### ISSUE TITLE
Plot: Clean up

### ISSUE THREAD

**Issue by bettinaselig** [cogneco/ooc-kean#800]
The Plot module leaks memory, has a lot of empty line and maybe also unnecessary code.

Some fixes where done in https://github.com/cogneco/ooc-kean/pull/663 and https://github.com/cogneco/ooc-kean/pull/664, which were closed.
@davidpiuva @marcusnaslund @davidhesselbom @sebastianbaginski

**Comment by bettinaselig** [comment]
In https://github.com/cogneco/ooc-kean/pull/805 empty lines and simplifies code.

@sebastianbaginski Can you take care of the memory leaks?

**Comment by sebastianbaginski** [comment]
Yes, I can do this if @davidpiuva has something else to do and I guess after #805 is merged.

**Comment by davidpiuva** [comment]
I have many things to do.

**Comment by sebastianbaginski** [comment]
https://github.com/cogneco/ooc-kean/issues/819
should we keep this one open ?

**Comment by davidhesselbom** [comment]
I think so. Take a look at any "plotter" anywhere - they are always a mess. The plotting library needs an overhaul.

**Comment by sebastianbaginski** [comment]
In that case I think it needs more detailed description.

**Comment by davidhesselbom** [comment]
Sure.

**Comment by davidpiuva** [comment]
I would like the plotter to be collection agnostic by just taking data as it comes with an add method inside a loop.

**Comment by davidhesselbom** [comment]
Not to mention how nice it would be to let the plotter automatically create lists from collections of types with more than one field...

**Comment by davidpiuva** [comment]
Bettina previously wanted the ability to sample values from seen points. Raw data output for visualizing applications might be useful when SVG is not enough.

**Comment by bettinaselig** [comment]
Not necessary, just nice feature.

**Comment by niklasborwell** [comment]
Since it's just for debugging, it seems like no one is giving this priority. This should probably be several more specific issues. Also it's stuck in the common "help wanted" limbo without conclusion. Suggest closing until someone cares about this again. @marcusnaslund @bettinaselig @sebastianbaginski @davidpiuva ?

**Comment by davidpiuva** [comment]
Maybe a milestone for "on ice"?

**Comment by niklasborwell** [comment]
If it is on ice I think we might as well close it. If everyone forgets about an issue for a long time it is probably not really an issue. If it reappears we can just make a new one.

**Comment by marcusnaslund** [comment]
> Maybe a milestone for "on ice"?

No, then they're closed.

**Comment by davidpiuva** [comment]
Then how do I separate old issues from solved issues if I want to solve something?

**Comment by bettinaselig** [comment]
I think it is ok to close it. The original issue was that there were memory leaks and empty lines. The empty lines are removed. The memory leaks still remain. But since it is only for debugging right now, it is not urgent in my eyes. 
If the memory leaks are a problem a new issue needs to be created. This one is out of date.

**Comment by marcusnaslund** [comment]
> Then how do I separate old issues from solved issues if I want to solve something?

It should be evident from how things work, or from the comments in the issue.

In any event, if there is something to be fixed, the issue should remain open but belong to a proper milestone.

**Comment by marcusnaslund** [comment]
> If the memory leaks are a problem a new issue needs to be created. This one is out of date.
